src/main/java/ulcrs/controllers/SessionController.java

```diff
@@ -14,36 +25,82 @@
 
 public class SessionController extends BaseController {
 
+    private static final String WORKSPACE_PATH = "workspace/";
+
     @Override
     public RouteGroup routes() {
         return () -> {
             before("/*", (request, response) -> log.info("endpoint: " + request.pathInfo()));
-            get("/", this::getSessionList, gson::toJson);
+            get("/", this::getSessionList, sessions -> {
+                Gson gson = new GsonBuilder().excludeFieldsWithoutExposeAnnotation().setPrettyPrinting().create();
+                return gson.toJson(sessions);
+            });
             get("/:name", this::getSession, gson::toJson);
-            post("/:name", this::updateSession, gson::toJson);
+            post("/:name", this::saveSession, gson::toJson);
             delete("/:name", this::deleteSession, gson::toJson);
         };
     }
 
     private List<String> getSessionList(Request request, Response response) {
-        // TODO: return names of sessions with limit specified
-        return null;
+        response.type(CONTENT_TYPE_JSON);
+        int limit = Integer.valueOf(request.params("limit"));
+
+        List<String> filenames = new ArrayList<>();
+        File folder = new File(WORKSPACE_PATH);
+        File[] listOfFiles = folder.listFiles();
+        if (listOfFiles != null) {
+            for (int i = 0; i < limit && i < listOfFiles.length; i++) {
+                File file = listOfFiles[i];
+                if (file.isFile()) {
+                    filenames.add(file.getName());
+                }
+            }
+        }
+        return filenames;
     }
 
     private Session getSession(Request request, Response response) {
-        // TODO: return session matching name
-        return null;
+        response.type(CONTENT_TYPE_JSON);
+        String name = request.params("name");
```

Comment:
Ditto here. How is this param sent from the frontend? Is it query param or body param?

Reply:
Here the name is in the path I believe, just like the id.